File: clients/go/model/city_cell.go
package model

import "io"
import . "komus24/stream"

// City cell
type CityCell int32

const (
    // Road
    CityCellRoad CityCell = 0
    // Building
    CityCellBuilding CityCell = 1
    // Refill station
    CityCellRefillStation CityCell = 2
)

// Read CityCell from reader
func ReadCityCell(reader io.Reader) CityCell {
    switch ReadInt32(reader) {
    case 0:
        return CityCellRoad
    case 1:
        return CityCellBuilding
    case 2:
        return CityCellRefillStation
    }
    panic("Unexpected tag value")
}

// Get string representation of CityCell
func CityCellToString(cityCell CityCell) string {
    switch cityCell {
    case CityCellRoad:
        return "Road"
    case CityCellBuilding:
        return "Building"
    case CityCellRefillStation:
        return "RefillStation"
    }
    panic("Impossible happened")
}

File: clients/csharp/Model/CityCell.cs
namespace Komus24.Model
{
    /// <summary>
    /// City cell
    /// </summary>
    public enum CityCell
    {
        /// <summary>
        /// Road
        /// </summary>
        Road = 0,
        /// <summary>
        /// Building
        /// </summary>
        Building = 1,
        /// <summary>
        /// Refill station
        /// </summary>
        RefillStation = 2,
    }

    public static class CityCellHelper {
        /// <summary> Read CityCell from reader </summary>
        public static CityCell ReadFrom(System.IO.BinaryReader reader) {
            switch (reader.ReadInt32())
            {
                case 0:
                    return CityCell.Road;
                case 1:
                    return CityCell.Building;
                case 2:
                    return CityCell.RefillStation;
                default:
                    throw new System.Exception("Unexpected tag value");
            }
        }
    }
}